### PR TITLE
Feat: add sidecar base url

### LINKED ISSUE #1280: Add support for sidecarBaseUrl
Now that we have a new regions file published with the `sidecarBaseUrl` field, we can remove hardcoded references to `"https://onebusaway.co"` and use that value instead. When `sidecarBaseUrl` is null, we will know that alerts, surveys, and notifications are not supported for that region.

This feature has already been added to the iOS in these two commits:

* https://github.com/OneBusAway/onebusaway-ios/pull/758/commits/833032b12e886ea09a371d61d784396509ac9002
* https://github.com/OneBusAway/onebusaway-ios/pull/758/commits/3edb6778312d17e08b8387d5d572191d711449d4

### PR DESCRIPTION
**Pull request by amrhossamdev**
Fixes #1280 

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `gradlew connectedObaGoogleDebugAndroidTest` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them for the initial submission of the pull request.  When addressing comments on a pull request, please push a new commit per comment when possible (reviewers will squash and merge using GitHub merge tool)

### CODE REVIEW

File: onebusaway-android/src/main/java/org/onebusaway/android/io/elements/ObaRegion.java [hunk 1]
```diff
@@ -67,6 +67,12 @@ public interface Open311Server {
      */
     public String getObaBaseUrl();
 
+    /**
+     * @return The sideCar Vase URL for this region, or null if it doesn't have a sideCar URL.
```
Comment (aaronbrethorst) [inline onebusaway-android/src/main/java/org/onebusaway/android/io/elements/ObaRegion.java:71]: small typo:

```
The sideCar Vase URL for this region, or null if it doesn't have a sideCar URL
```

->

```
The sidecar base URL for this region, or null if it doesn't have a sidecar URL
```

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
looking good! I'd like to see one comment changed and then there's a test failure to resolve.

**Comment by amrhossamdev** [comment]
I removed the region weather test as it was primarily testing the IO functionality. I believe it's not essential since it requires a sidecar URL, which is not mandatory. Let me know your thoughts. @aaronbrethorst

**Review by aaronbrethorst — APPROVED** [review]
lgtm 👍

**Pull request merged**